Add "Save Deletion Options" in Setting Menu QoL Feature.

Glought opened the pull request:
Save Deletion Options:
* Add "Saves" Text Separator.
* Add "Enable Save Deletion Options" Checkbox.
* Add "Delete All Saves" Button.
* Add "Select Save File To Delete" Slider.
* Add "Delete Selected Save File" Button.

The save deletion options  are in "Settings -> General -> General Settings" under the "Open App Files Folder" button.

The "Enable Save Deletion Options" Checkbox when checked enables the save deletion options below it. It is uncheck by default.

The purpose for the "Enable Save Deletion Options" checkbox is to prevent the possibility of accidentally pressing the "Delete All Saves" or "Delete Selected Save File " Button by mistake.

When the "Delete All Saves" Button is pressed it will delete all 3 save files and resets SoH and then disables itself by turning off "Enable Save Deletion Options" Checkbox.

The reasoning for this feature is to have a quick way to delete all saves files instead of deleting each file using the in-game file menu(takes a while took 30 actions to delete all 3 files) or navigating the App file folder(Would require closing Soh, the reset button doesn't clear the files slots in-game if files are deleted outside of SoH.).

The user can also use the "Select Save File To Delete" Slider to delete a specific save file then press
the "Delete Selected Save File" Button to delete the selected file.
It will disable itself and reset SoH.

<img width="1021" height="849" alt="Enable_Deletion_Options_Git_pic1" src="https://github.com/user-attachments/assets/fb86b9ab-3897-4829-9b3e-b5faf2a00016" />


A use case for this feature, Like if a Rando run isn't going well and you want to start over just open the Settings menu
and delete your save like shown in this video.


https://github.com/user-attachments/assets/fcda45f9-1dfd-4b35-bb98-a2a2d8413182



<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3625074694.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3625077321.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3625084376.zip)
<!--- section:artifacts:end -->

Glought committed:
Add "Delete All Saves" QoL Feature.
*Added "Enable Delete All Saves" Checkbox.
*Added "Delete All Saves" Button

Both are in "Settings -> General -> General Settings" under the
"Open App Files Folder" button.

The "Enable Delete All Saves" Checkbox when checked enables the
"Delete All Saves" Button.It is uncheck by default.

The purpose for the "Enable Delete All Saves" checkbox is to prevent the
possibility of accidentally pressing the "Delete All Saves" Button by
mistake.

When the "Delete All Saves" Button is pressed it will delete all 3 save
files and resets SoH and then disables itself by turning off
"Enable Delete All Saves" Checkbox.

The Reasoning for this Qol is to have a quick way to
delete all saves instead of deleting each file using the in-game
file menu(takes a while took 30 actions to delete all 3 files)
or navigating the App file folder(Would require closing Soh, the
reset button doesn't clear the files slots in-game
if files are deleted outside of SoH.).

serprex commented on soh/soh/SohGui/SohMenuSettings.cpp:
```diff
@@ -187,6 +187,28 @@ void SohMenu::AddMenuSettings() {
+                "This button is disabled because \"Enable Delete All Saves\" is turned off.";
+        })
+        .Callback([](WidgetInfo& info) {
+            GameInteractor::Instance->ExecuteHooks<GameInteractor::OnDeleteAllFiles>();
```
shouldn't need to add hook, should be able to call `DeleteZeldaFile(fileNum)` loop here & that'll kick off existing delete file hooks

Glought replied:
I tried that it required #include the SaveManager.h into the SohMenuSettings.cpp kinda didn't want to #include it just use one function unless that's fine to do.

serprex replied:
That's fine to do. You could also copy the function prototype if you wanted (tho others might disagree & I'm on the fence there)

Glought replied:
Thank you for your feedback 👍 I reworked it to no longer need the hooks I added.

Glought committed:
Add "Delete Selected Save file" in Settings -> General Menu
* Add "Saves" Text Separator.
* Renamed "Enable Delete All Saves" checkbox to "Enable Delete All/Selected Saves".
* Add "Select Save File To Delete" Slider.
* Add "Delete Selected Save File" Button.

The User can use the Slider to select the file to delete then press
the "Delete Selected Save File" Button to delete the selected file.
It will Disable itself and reset SoH.

Glought committed:
Removed the need for the "OnDeleteSelectedFile and "OnDeleteAllFiles Hooks.
Reworked it to use the SaveManger's "DeleteZeldaFile()" function in the
"Delete Selected Save File" and "Delete All Saves" button callback.

Glought committed:
Renamed "Enable Delete All/Selected Saves" Checkbox to "Enable Save Deletion Options"
Renamed the "EnableDeleteAllSelectedSaveFiles" CVar to "EnableSaveDeletionOptions"

Glought commented:
I changed the Pull Request Name to "Add "Save Deletion Options" in Setting Menu QoL Feature." felt like a fitting name now that i expanded from just "Deleting All Save Files" to being able to Delete Selected save files from the Setting Menu.

Malkierian commented:
We had a discussion on the dev team, and we generally agree that this feels like too much added for barely any perceptible gain, so I'm going to go ahead and close it. The reason is because the only thing it really adds is the Delete All Saves functionality, but I can't see that use case being common at all. The individual file deletion feels like it takes the same amount of time to complete, just in the SoH menu instead of File Select. The code generally looked clean, though, so I look forward to any future contributions you might feel inclined to submit.
